FastCopyShellExtension/Recorder.cpp

```
@@ -1,84 +1,174 @@
+    size_t const length = buf.size();
+    fwrite(&length, sizeof(length), 1, m_fs);
+    fwrite(buf.data(), 2, length, m_fs);
 
-Recorder::~Recorder()
-{
-	if (m_fs)
-		fclose(m_fs);
+    return *this;
 }
 
-static wchar_t toFlag(CopyOperation op)
+Recorder::~Recorder()
 {
-	switch (op)
-	{
-		case CopyOperation::Copy:
-			return L'C';
-		case CopyOperation::Move:
-			return L'M';
-		case CopyOperation::Paste:
-			return L'P';
-		case CopyOperation::Delete:
-			return L'D';
-	}
+    if (m_fs)
+        fclose(m_fs);
 }
 
 std::filesystem::path Recorder::GetRecordFilePath(CopyOperation op)
 {
-	return 	GetLocalDataFolder() / std::format(L"{}{}.txt", toFlag(op), GetTimeString());
+    auto& s = FastCopy::Settings::CommonSharedSettings::Instance();
+    auto const& folderOpt = s.GetLocalDataFolder();
+    if (!folderOpt)
+    {
+        FC_LOG_ERROR(L"LocalDataFolder unavailable, skip recording.");
+        return {};
+    }
+
+    auto const& folder = *folderOpt;
+    return folder / std::format(L"{}{}.txt", toFlag(op), GetTimeString());
 }
 
 bool Recorder::HasRecord()
 {
-	return std::find_if(std::filesystem::directory_iterator{ GetLocalDataFolder() }, std::filesystem::directory_iterator{}, [](std::filesystem::directory_entry const& fileEntry) {
-		auto str = fileEntry.path().filename().wstring();
-		return str.starts_with(L'C') || str.starts_with(L"M2") || str.starts_with(L'P');
-	}) != std::filesystem::directory_iterator{};
+    try
+    {
+        auto& s = FastCopy::Settings::CommonSharedSettings::Instance();
+        auto const& folderOpt = s.GetLocalDataFolder();
+        if (!folderOpt) {
+            FC_LOG_ERROR(L"LocalDataFolder unavailable, skip recording.");
+            return false;
+        }
+
+        auto const& folder = *folderOpt;
+
+        if (folder.empty())
+        {
+            FC_LOG_ERROR(L"LocalDataFolder empty, skip recording.");
+            return false;
+        }
+
+        std::error_code ec;
```

Not necessary either. Just let it throw